File: frontend/src/types/company.ts
// ── KPI Colors types ───────────────────────────────────
export interface KpiThreshold {
  min?: number | null;
  max?: number | null;
}

export interface KpiColorEntry {
  green: KpiThreshold;
  yellow: KpiThreshold;
  red: KpiThreshold;
}

export interface KpiColorsConfig {
  roas: KpiColorEntry | null;
  cpa: KpiColorEntry | null;
  ctr: KpiColorEntry | null;
  cpc: KpiColorEntry | null;
}

// ── AI Instructions types ──────────────────────────────
export interface MetricRule {
  good: string;
  bad: string;
  average: string;
}

export interface AiMetricsConfig {
  roas: MetricRule | null;
  cpa: MetricRule | null;
  cpc: MetricRule | null;
  connect_rate: MetricRule | null;
}

export interface AiInstructions {
  metrics: AiMetricsConfig;
  additional_prompt: string;
}

// ── Company types ──────────────────────────────────────

export interface OperationalCost {
  id: string;
  name: string;
  amount: number;
}

export interface CompanySettings {
  tax_rate: number;
  operational_costs: OperationalCost[];
}

export interface MonthlyFinancial {
  month: string;
  label: string;
  revenue: number;
  losses: number;
  spend: number;
  profit: number;
}

export interface CompanyDashboard {
  year: number;
  monthly: MonthlyFinancial[];
  total_sales: number;
  unique_customers: number;
}

File: backend/database/models/company.py
from sqlalchemy import Column, Integer, Float, JSON, Text, DateTime, Boolean
from database.core.connection import Base
from database.core.timezone import UPDATED_AT_DEFAULT


DEFAULT_KPI_COLORS = {
    "roas": {
        "green": {"min": 3},
        "yellow": {"min": 2, "max": 3},
        "red": {"max": 2},
    },
    "cpa": None,
    "ctr": None,
    "cpc": None,
}

DEFAULT_AI_INSTRUCTIONS = {
    "metrics": {
        "roas": None,
        "cpa": None,
        "cpc": None,
        "connect_rate": None,
    },
    "additional_prompt": "",
}


class CompanySettings(Base):
    __tablename__ = "company_settings"

    id = Column(Integer, primary_key=True, autoincrement=True)
    tax_rate = Column(Float, default=12.3)
    operational_costs = Column(JSON, default=list)
    kpi_colors = Column(JSON, default=lambda: DEFAULT_KPI_COLORS.copy())
    ai_instructions = Column(JSON, default=lambda: DEFAULT_AI_INSTRUCTIONS.copy())
    stripe_enabled = Column(Boolean, default=False, nullable=False, server_default="false")
    updated_at = Column(
        DateTime,
        server_default=UPDATED_AT_DEFAULT,
        onupdate=UPDATED_AT_DEFAULT,
    )
